access/validator.go

```diff
@@ -141,6 +143,12 @@ func (v *TransactionValidator) checkMissingFields(tx *flow.TransactionBody) erro
 }
 
 func (v *TransactionValidator) checkGasLimit(tx *flow.TransactionBody) error {
+	// if service account is the payer of the transaction accepts any gas limit
+	// note that even though we don't enforce any limit here, exec node later
+	// enforce a max value for any transaction
+	if tx.Payer == v.serviceAccountAddress {
```

Comment:
any reason we didn't just check against `chain.ServiceAddress()` here? I'm guessing saving that function call?

Reply:
yeah not a cheap function to be called for every transaction, it creates a new address generator and asks for the first address.